ClassRegistration/ClassRegistration.App/Controllers/EnrollmentController.cs

```diff
@@ -0,0 +1,89 @@
+﻿
+using ClassRegistration.App.Interfaces;
+using ClassRegistration.App.Repositories;
+using ClassRegistration.DataAccess.Entity;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Security.Cryptography.X509Certificates;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace ClassRegistration.App.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class EnrollmentController : ControllerBase
+    {
+
+        /// <summary>
+        /// private IEnrollment field.
+        /// </summary>
+        private IEnrollmentRepo _enrollmentRepository;
+
+        
+        /// <summary>
+        /// Constructor initializes IEnrollment field.
+        /// </summary>
+        /// <param name="enrollmentRepo"></param>
+        public EnrollmentController(IEnrollmentRepo enrollmentRepo)
+        {
+            this._enrollmentRepository = enrollmentRepo;
+
+        }
+        
+       
+
+        /// <summary>
+        /// This method returns the total credits of a student with a specified ID and the term
+        /// </summary>
+        /// <param name="id"></param>
+        /// <param name="term"></param>
+        /// <returns></returns>
+
+        // GET api/<EnrollmentController>/id
+        [HttpGet("{id}/{term}")]
+        public Object GetTotalCredits(int id,string term)

```

Comment:
Return type should be "IActionResult". Also, we'll want to make controller and repository methods async

Reply:
Resolved.